textacy/fileio/read.py

```diff
@@ -0,0 +1,187 @@
+
+                {"title": "Harrison Bergeron", "text": "The year was 2081, and everybody was finally equal."}\n
+                {"title": "2BR02B", "text": "Everything was perfectly swell."}
+
+        mode (str, optional)
+        encoding (str, optional)
+
+    Yields:
+        dict: next valid JSON object, converted to native Python equivalent
+    """
+    with io.open(filename, mode=mode, encoding=encoding) as f:
+        for line in f:
+            yield json.loads(line)
+
+
+def read_json_mash(filename, mode='rt', encoding=None, buffersize=2048):
+    """
+    Iterate over a stream of JSON objects, all of them mashed together, end-to-end,
+    on a single line of a file. Bad form, but still manageable.
+
+    Args:
+        filename (str): /path/to/file on disk from which json objects will be streamed,
+            where all json objects are mashed together, end-to-end, on a single line,;
+            for example::
+
+                {"title": "Harrison Bergeron", "text": "The year was 2081, and everybody was finally equal."}{"title": "2BR02B", "text": "Everything was perfectly swell."}
+
+        mode (str, optional)
+        encoding (str, optional)
+        buffersize (int, optional): number of bytes to read in as a chunk
+
+    Yields:
+        dict: next valid JSON object, converted to native Python equivalent
+    """
+    with io.open(filename, mode=mode, encoding=encoding) as f:
+        buffer = ''
+        for chunk in iter(partial(f.read, buffersize), ''):
+            buffer += chunk
+            while buffer:
+                try:
+                    result, index = JSON_DECODER.raw_decode(buffer)
+                    yield result
+                    buffer = buffer[index:]
+                # not enough data to decode => read another chunk
+                except ValueError:
+                    break
+
+
+def read_file(filename, mode='rt', encoding=None):
+    """
+    Read the full contents of a file. Files compressed with gzip or bz2 are handled
+    automatically.
+    """
+    _open = gzip.open if filename.endswith('.gz') \
+        else bz2.open if filename.endswith('.bz2') \
+        else io.open
+    with _open(filename, mode=mode, encoding=encoding) as f:
+        return f.read()
+
+
+def read_file_lines(filename, mode='rt', encoding=None):
+    """
+    Read the contents of a file, line by line. Files compressed with gzip or bz2
+    are handled automatically.
+    """
+    _open = gzip.open if filename.endswith('.gz') \
+        else bz2.open if filename.endswith('.bz2') \
+        else io.open
+    with _open(filename, mode=mode, encoding=encoding) as f:
+        for line in f:
+            yield f
+
+
+def read_spacy_docs(spacy_vocab, filename):
+    """
+    Stream ``spacy.Doc`` s from disk at ``filename`` where they were serialized
+    using Spacy's ``spacy.Doc.to_bytes()`` functionality.
+
+    Args:
+        spacy_vocab (``spacy.Vocab``): the spacy vocab object used to serialize
```

Comment:
I'm not sure how often Spacy's language vocab changes, or if people use custom vocabs of their own — [this](https://github.com/spacy-io/spaCy/blob/master/spacy/vocab.pyx#L51) vs [this](https://github.com/spacy-io/spaCy/blob/master/spacy/vocab.pyx#L55). It would really help if spacy had more documentation throughout its code base...